File: tests/csharp/csharp_error_handling/nested_try_catch.cs
// vybe-test: csharp/csharp_error_handling/nested_try_catch
// origin: languages/csharp/tests/csharp/test_csharp_error_handling.rs

using static __Harness;

try {
    try {
        throw new Exception("inner");
    } catch (Exception e) {
        __P(("inner: " + e.Message).ToString());
        throw new Exception("rethrown");
    }
}
catch (Exception e) {
    __P(("outer: " + e.Message).ToString());
}
__Check("inner: inner\nouter: rethrown");

public static class __Harness {
    public static string __buf = "";
    public static void __P(string s) { __buf = __buf + s + "\n"; }
    public static void __Pr(string s) { __buf = __buf + s; }
    public static void __Check(string want) {
        if (__buf != want && __buf != want + "\n") {
            Console.WriteLine("FAIL: want [" + want + "] got [" + __buf + "]");
            throw new Exception("assertion failed");
        }
    }
}

File: tests/js/ecma_error_handling/nested_try_catch.js
// vybe-test: js/ecma_error_handling/nested_try_catch
// origin: languages/js/tests/js/test_ecma_error_handling.rs

function __line(...args) {
    // console.log joins its arguments with a single space. String() is the
    // coercion Vybe's logging host applies to each one.
    return args.map(String).join(" ");
}

function __check(got, want) {
    if (got !== want) {
        console.log("FAIL: want [" + want + "] got [" + got + "]");
        throw new Error("assertion failed");
    }
}

try {
    try {
        throw new Error("inner");
    } catch (e) {
        __check(__line("inner: " + e.message), "inner: inner");
        throw new Error("rethrown");
    }
} catch (e) {
    __check(__line("outer: " + e.message), "outer: rethrown");
}
